src/lib/reducers/content.js

```diff
@@ -15,15 +15,25 @@ const defaultState = {
 }
 
 export const reducer = (state = defaultState, action) => {
-  if (
-    action.type === SET_CONTENT &&
-    state[action.location] !== action.content
-  ) {
-    return { ...state, [action.location]: action.content }
+  if (action.type === SET_CONTENT) {
+    if (!action.location) return state
+    const newState = Array.isArray(state[action.location])
+      ? state[action.location].slice()
+      : []
+    // avoid doublon
+    if (
+      newState[newState.length - 1] &&
+      newState[newState.length - 1] === action.content
+    ) {
+      return state
+    }
+    newState.push(action.content)
+    return { ...state, [action.location]: newState }
   } else {
     return state
   }
 }
 
 // selectors
-export const getContent = (state, location) => state[location]
+export const getContent = (state, location) =>
+  Array.isArray(state[location]) ? state[location].pop() : null
```

Comment:
C'est surprenant qu'une méthode appelée `get` ait un side effect.

Reply:
It will be better with a `unsetContent` handled by the reducer itself, and this function will be without side effect